pkg/network/master/egressip.go

```diff
@@ -170,6 +175,19 @@ func (eim *egressIPManager) check(retrying bool) bool {
 			continue
 		}
 
+		nn, err := eim.nodeInformer.Lister().Get(node.name)
+		if err != nil {
+			return err
+		}
+
+		for _, cond := range nn.Status.Conditions {
+			if cond.Type == v1.TaintNodeNotReady {
+				klog.Warningf("Node %s is offline", node.ip)
+				node.offline = true
+				eim.tracker.SetNodeOffline(node.ip, true)
```

Comment:
I think you need to return here, or the node will go back online if the ping succeeds.